wandb/sdk/launch/launch.py

```diff
@@ -46,17 +59,19 @@ def resolve_agent_config(
     }
     user_set_project = False
     resolved_config: Dict[str, Any] = defaults
-    if os.path.exists(os.path.expanduser(LAUNCH_CONFIG_FILE)):
-        config = {}
-        with open(os.path.expanduser(LAUNCH_CONFIG_FILE)) as f:
+    config_path = config or os.path.expanduser(LAUNCH_CONFIG_FILE)
+    if os.path.isfile(config_path):
+        launch_config = {}
+        with open(config_path) as f:
             try:
-                config = yaml.safe_load(f)
-                print(config)
+                launch_config = yaml.safe_load(f)
             except yaml.YAMLError as e:
                 raise LaunchError(f"Invalid launch agent config: {e}")
-        if config.get("project") is not None:
+        if launch_config.get("project") is not None:
             user_set_project = True
-        resolved_config.update(dict(config))
+        resolved_config.update(launch_config.items())
+    elif config is not None:
+        raise LaunchError(f"Could not find config file: {config_path}")
```

Comment:
I think we should just default to a local env, local registry and docker builder if this doesn't exist, so users don't need to set up the config file when they try launch for the first time.

Reply:
This is only hit if they specify a config file via the cli arg and that file doesn't exist. I'm going to update the message to indicate this. The number of variables in this function with "config" in the name is tough.